transports/tls/tls.go

```diff
@@ -22,9 +53,101 @@ func rawTLSAdapter(route *router.Route) (router.LogAdapter, error) {
 type tlsTransport int
 
 func (t *tlsTransport) Dial(addr string, options map[string]string) (net.Conn, error) {
-	conn, err := tls.Dial("tcp", addr, nil)
+
+	// at this point, if our trust store is empty, there is no point of continuing...
+	if len(clientTLSConfig.RootCAs.Subjects()) < 1 {
+		return nil, fmt.Errorf("FATAL: TLS CA trust store is empty! Can not trust any TLS endpoints: tls://%s", addr)
+	}
+
+	// attempt to establish the TLS connection
+	conn, err := tls.Dial("tcp", addr, clientTLSConfig)
 	if err != nil {
 		return nil, err
 	}
 	return conn, nil
 }
+
+// createTLSConfig creates the required TLS configuration that we need to establish a TLS connection
+func createTLSConfig() (*tls.Config, error) {
+
+	var err error
+	tlsConfig := &tls.Config{}
+
+	// use stronger TLS settings if enabled
+	// TODO: perhaps this should be default setting
+	if os.Getenv(envTLSHardening) != "true" {
```

Comment:
please forgive my naivety but what are these set to if `envTLSHardening` is `true`?

Reply:
Nice catch, the condition should have read `==`. I committed the fix

Reply:
Can you make sure the tests cover both configurations?